### ISSUE TITLE
gsutil broken by upgrade to pyopenssl (24.3)

### ISSUE THREAD

**Issue by phlax** [GoogleCloudPlatform/gsutil#1820]
pyopenssl has removed a load of libs and this now breaks gsutil (importing `crypto` from pyopenssl)

probably the dep here needs to be pinned pending an update to the code

**Comment by sameer-google** [comment]
@phlax, The recent release of `pyOpenSSL` on `2024-11-28`, adding support for `Python 3.13` and removing deprecated modules, could indeed be causing the issue. Could you provide more details about what breaks? Specifically, any `error logs` or details about the affected functionality would be helpful in confirming whether this version is the root cause and identifying the exact issue.

**Comment by phlax** [comment]
if you see here the failing CI:

https://github.com/envoyproxy/envoy/actions/runs/12067084022/job/33649308904

on arm machines where we need to use gsutil - its not installed on the github-hosted runners so we do (or did):

```console
pip install gsutil
```

then doing `gsutil sync` we get:

```console
module 'OpenSSL.crypto' has no attribute 'sign'
```

thinking about it now it might be related to the `crcmod` binary - so maybe less broken than i originally thought - but at least. `sync` on arm is broken with a pip install

**Comment by phlax** [comment]
actually not `sync` - its a `cp` command - so possibly just arm

**Comment by jayhatha** [comment]
Same error here. `gsutil cp` inside a Debian 11 container in GitHub actions (so, running on AMD64).

**Comment by alex** [comment]
Is it possible to exact the _full_ traceback?

**Comment by spiralman** [comment]
@sushantkhanna and @sameer-google As reported in the original PR, #1821, which was closed, the fix you merged in #1822 does *not* resolve this issue, at least not in the official google/gcloud-sdk Docker images. Would it be possible to ensure the issue is fixed in the official docker images?

**Comment by sameer-google** [comment]
Alpine images is rolled back to version `3.19` in `504.0.0-alpine`. So the issue with `OpenSSL.crypto` should be resolved now.

Reference: [cloud-sdk-docker/issues/497](https://github.com/GoogleCloudPlatform/cloud-sdk-docker/issues/497)
